presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionarySelectiveReader.java

```diff
@@ -365,33 +386,69 @@ public Block getBlock(int[] positions, int positionCount)
         checkState(positionCount <= outputPositionCount, "Not enough values");
         checkState(!valuesInUse, "BlockLease hasn't been closed yet");
 
-        if (allNulls) {
+        if (allNulls || nullsCount == outputPositionCount) {
```

Comment:
curious why just ```allNulls``` check won't be sufficient here

Reply:
allNulls is set true when `dataStream == null && presentStream != null`. When this condition is false, it's still possible that all the selected positions after read() is finished are null. Think of the starting batches where `batchSize == 1` or some small numbers. It's possible this value is null.